src/BlazorWasmSdk/Tasks/ComputeBlazorBuildAssets.cs

```diff
@@ -129,6 +128,14 @@ public override bool Execute()
                         candidate.SetMetadata("RelativePath", $"_framework/{destinationSubPath}");
                     }
 
+                    // Workaround for https://github.com/dotnet/aspnetcore/issues/37574.
+                    // For items added as "Reference" in project references, the OriginalItemSpec is incorrect.
+                    // Ignore it, and use the FullPath instead.
+                    if (candidate.GetMetadata("ReferenceSourceTarget") == "ProjectReference")
```

Comment:
`ProjectReference` is this the value that `Reference` items set it to? What happens with dlls that come from `ProjectReference` do they also have this property set up with the same value?

Reply:
Both the referenced dll and the app dll have this value. The way MSBuild constructs the Item for the reference is by taking the project's output and updating it's ItemSpec, so it doesn't have any defining properties to differentiate on.